Lib/asyncore.py

```diff
@@ -399,6 +403,9 @@ def close(self):
                 if why.args[0] not in (ENOTCONN, EBADF):
                     raise
 
+    def fileno(self):
+        return self.socket.fileno()
```

Comment:
Looks like that in close `self.fd` is only set to `-1` in python3, in python2 just `os.close(self.fd)` is called. The behaviour for regular file object when `fileno()` is called on them is to raise
```python
>>> f.close()
>>> f.fileno()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: I/O operation on closed file
```
would raising this be a possible solution?